snooty/parser.py

```diff
@@ -484,6 +497,27 @@ def validate_doc_role(self, node: docutils.nodes.Node) -> None:
             msg = f'"{node["name"]}" could not open "{resolved_target_path}": No such file exists'
             self.diagnostics.append(Diagnostic.error(msg, util.get_line(node)))
 
+    def validate_list_table(
+        self, node: docutils.nodes.Node, expected_num_columns: int
+    ) -> None:
+        """Validate list-table structure"""
+        if (
+            isinstance(node, docutils.nodes.bullet_list)
+            and len(node.children) != expected_num_columns
+        ):
+            msg = (
+                f'expected "{expected_num_columns}" columns, saw "{len(node.children)}"'
+            )
+            self.diagnostics.append(
+                Diagnostic.error(msg, util.get_line(node) + len(node.children) - 1)
+            )
+            return
+
+        # unwrap the data structures holding the bullet list
+        if not isinstance(node, docutils.nodes.bullet_list):
+            for child in node.children:
+                self.validate_list_table(child, expected_num_columns)
+
```

Comment:
I really don't think we need to recurse here, and could delete this code entirely. Recursing is redundant: `validate_list_table()` will be called automatically on every `list-table` directive, and we don't need to run this validation on any other type of node that appears as a list table's child. Let me know if I can clarify this for you further, it's definitely a bit confusing!

Reply:
Oh I was actually using this recursive call to remove the containers around the first `bullet_list`, but I just realized that I could've just done it in a simpler way without recursion 😄. Adding that soon!